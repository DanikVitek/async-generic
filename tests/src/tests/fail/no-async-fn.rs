use async_generic::async_generic;

#[async_generic]
async fn do_some_stuff() -> bool {
    // ERROR: async should not be specified on an async_generic fn.
    true
}

fn main() {}
