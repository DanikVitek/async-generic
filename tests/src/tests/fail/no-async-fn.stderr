error: async_generic can only be used with functions, traits or impls
 --> src/tests/fail/no-async-fn.rs:3:1
  |
3 | #[async_generic]
  | ^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `async_generic` (in Nightly builds, run with -Z macro-backtrace for more info)

error: an async_generic function should not be declared as async
 --> src/tests/fail/no-async-fn.rs:4:1
  |
4 | async fn do_some_stuff() -> bool {
  | ^^^^^

error: expected `trait`
 --> src/tests/fail/no-async-fn.rs:4:1
  |
4 | async fn do_some_stuff() -> bool {
  | ^^^^^

error: expected `impl`
 --> src/tests/fail/no-async-fn.rs:4:1
  |
4 | async fn do_some_stuff() -> bool {
  | ^^^^^
